### ISSUE TITLE
Error when executing tensorflow model in gocv/ How to prepare tensorflow models to run with gocv?

### ISSUE THREAD

**Issue by twatzl** [hybridgroup/gocv#474]
## Description
So I want to use gocv to classify using different pretrained tensorflow models. For that I downloaded several models from https://github.com/tensorflow/models/tree/master/research/slim and also checked out the repo.

I then created the .pb files using the command described in the slim readme:
```
python export_inference_graph.py \
  --alsologtostderr \
  --model_name=inception_v3 \
  --output_file=/tmp/inception_v3_inf_graph.pb
```
However executing using this model resulted in an error about some variables that could not be found, so I used this command to freeze the graph (the paths on my system were a bit different):

```
bazel-bin/tensorflow/python/tools/freeze_graph \
  --input_graph=/tmp/inception_v3_inf_graph.pb \
  --input_checkpoint=/tmp/checkpoints/inception_v3.ckpt \
  --input_binary=true --output_graph=/tmp/frozen_inception_v3.pb \
  --output_node_names=InceptionV3/Predictions/Reshape_1
```
So I had the frozen model, but when I run the gocv code with it it results in the following error:

```
 go run ./main.go ./pics/ ./model/optimized_inception_v3.pb ./model/imagenet_comp_graph_label_strings.txt opencv cpu
Start classifying 6 pictures.
terminate called after throwing an instance of 'cv::Exception'
  what():  OpenCV(4.1.0) /home/myuser/opencv-4.1.0/modules/dnn/src/dnn.cpp:524: error: (-2:Unspecified error) Can't create layer "InceptionV3/InceptionV3/Conv2d_1a_3x3/BatchNorm/FusedBatchNormV3" of type "FusedBatchNormV3" in function 'getLayerInstance'

SIGABRT: abort
PC=0x7ff2289ea53f m=0 sigcode=18446744073709551610
signal arrived during cgo execution

goroutine 1 [syscall, locked to thread]:
runtime.cgocall(0x4bbf90, 0xc000044cd0, 0xc000044ce0)
        /usr/local/go/src/runtime/cgocall.go:128 +0x5b fp=0xc000044ca0 sp=0xc000044c68 pc=0x41cfab
gocv.io/x/gocv._Cfunc_Net_Forward(0x24bd360, 0x25289a0, 0x0)
        _cgo_gotypes.go:4360 +0x4e fp=0xc000044cd0 sp=0xc000044ca0 pc=0x4b696e
gocv.io/x/gocv.(*Net).Forward.func2(0xc000044e40, 0x25289a0, 0x25289a0)
        /home/myuser/go/src/gocv.io/x/gocv/dnn.go:151 +0x68 fp=0xc000044d10 sp=0xc000044cd0 pc=0x4b8598
gocv.io/x/gocv.(*Net).Forward(0xc000044e40, 0x505d45, 0x8, 0x0)
        /home/myuser/go/src/gocv.io/x/gocv/dnn.go:151 +0x7d fp=0xc000044d40 sp=0xc000044d10 pc=0x4b765d
main.main()
        /home/myuser/go/src/github.com/myuser/img-classifier-gocv/main.go:112 +0x64d fp=0xc000044f98 sp=0xc000044d40 pc=0x4b9add
runtime.main()
        /usr/local/go/src/runtime/proc.go:200 +0x20c fp=0xc000044fe0 sp=0xc000044f98 pc=0x44429c
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:1337 +0x1 fp=0xc000044fe8 sp=0xc000044fe0 pc=0x46bb21

rax    0x0
rbx    0x6
rcx    0x7ff2289ea53f
rdx    0x0
rdi    0x2
rsi    0x7ffda3f55c90
rbp    0x7ff228b73860
rsp    0x7ffda3f55c90
r8     0x0
r9     0x7ffda3f55c90
r10    0x8
r11    0x246
r12    0x25a3640
r13    0x252afa0
r14    0x252afc0
r15    0x20c
rip    0x7ff2289ea53f
rflags 0x246
cs     0x33
fs     0x0
gs     0x0
exit status 2
```

I also tried using the `optimize_for_inference.py` hoping that it would remove that layer, but it did not change anything in the error message.

Now I ran out of ideas how to fix this issue. I am not sure if the way to obtain the tensorflow models is the correct one, but I could not find any other way. My guess would be that I am missing some crucial step.

How can I prepare tensorflow models so that they can be used using gocv?

## Your Environment

* Operating System and version: **Fedora 29 64-bit (Linux 4.20.16-200.fc29.x86_64)** 
* OpenCV version used: **4.1.0**
* How did you install OpenCV? `make install`
* GoCV version used: **could not find out version**
* Go version: **go1.12 linux/amd64**
* Did you run the `env.sh` or `env.cmd` script before trying to `go run` or `go build`?  **no**

**Comment by WendelHime** [comment]
Hi @twatzl , seems like you've found a layer which isn't implemented yet (FusedBatchNormV3). 

>  The first solution is to create a feature request at https://github.com/opencv/opencv/issues mentioning details such a source of model and type of new layer. A new layer could be implemented if OpenCV community shares this need.

[This link](https://docs.opencv.org/master/dc/db1/tutorial_dnn_custom_layers.html) show how to implement a custom layer if you really really want to make it work.

**Comment by twatzl** [comment]
Ok. Thanks.

If I implement this layer, can I be sure that it is the only one missing or is it possible that there are further unsupported layers?

**Comment by WendelHime** [comment]
Nop, you can check if the layers are implemented [in this partial list](https://docs.opencv.org/3.4/d6/d87/group__dnnLayerList.html)

**Comment by twatzl** [comment]
So after a week of research and headache I think I found the problem.
InceptionV3 uses the `batch_norm` function from tensorflow.contrib. However contrib will be discontinued and somehow there is also no libraries (.so files) included anymore for the contrib functions.

I did not only have this problem in gocv, but also with the official go bindings for tensorflow and with tensorflow/serving. So it seems to be a problem of the InceptionV3 model an not of gocv.
